packages/core/saved-objects/core-saved-objects-import-export-server-internal/src/import/lib/validate_references.ts

```diff
@@ -102,30 +102,35 @@ export async function validateReferences(params: ValidateReferencesParams) {
   const nonExistingReferenceKeys = await getNonExistingReferenceAsKeys(params);
 
   // Filter out objects with missing references, add to error object
-  objects.forEach(({ type, id, references, attributes }) => {
-    if (objectsToSkip.has(`${type}:${id}`)) {
+  for (const obj of objects) {
+    const { type, id, references, attributes } = obj;
+    const objectKey = `${type}:${id}`;
+    if (objectsToSkip.has(objectKey)) {
       // skip objects with retries that have specified `ignoreMissingReferences`
-      return;
+      continue;
     }
 
     const missingReferences = [];
```

Comment:
Retro nit: I know this was here before, but we should give this a type rather than default to any[]

Reply:
Updated in: https://github.com/elastic/kibana/pull/200053/commits/5544b8874f239262db3185604fd23b998e33b175